library/entity/status_effect/src/main/java/org/quiltmc/qsl/entity/effect/api/StatusEffectRemovalReason.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.entity.effect.api;
+
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.effect.StatusEffect;
+import net.minecraft.entity.effect.StatusEffectInstance;
+import net.minecraft.util.Identifier;
+
+/**
+ * Represents a reason for a {@link StatusEffectInstance} to be removed.
+ */
+@SuppressWarnings("ClassCanBeRecord")
```

Comment:
Hmm, I'm kinda wary of those weird CamelCase SupressWarnings around the code; Are they IntelliJ-only or something?

Reply:
Yeah, that's an IntelliJ thing. It'll yell at you to make the class a record if they're not there. It *shouldn't* affect anything on IDEs that don't support that particular warning.

Reply:
IntelliJ really wants this class to be a record purely because I don't want to mark it as `abstract`.